piptools/utils.py

```diff
@@ -115,7 +116,16 @@ def format_requirement(
     if ireq.editable:
         line = f"-e {ireq.link.url}"
     elif is_url_requirement(ireq):
-        line = ireq.link.url
+        # if the requirement has no name then it's just a URL
+        if not ireq.name:
```

Comment:
For `if-else` statements I would recommend the following best practice:

> Put the case you normally expect to process first. This is in line with the general principle of putting code that results from a decision as close as possible to the decision.

```python
if ok:
    do_ok_stuff()
else:
    do_not_ok_stuff()
```

Reply:
Thanks for your feedback, I've updated the logic.